backend/src/controllers/UserController.ts

```diff
@@ -1,45 +1,60 @@
 import asyncHandler from 'express-async-handler';
-import { Request, Response } from 'express';
-import { DevLoginRequest, LoginRequest } from '../types/RequestTypes';
-import User, { UserModel } from '../models/UserModel';
+import User, { UserDocument, UserPopulatedDocument } from '../models/UserModel';
 import jwt from 'jsonwebtoken';
 import config from '../types/Config';
 import { OAuth2Client } from 'google-auth-library';
-import Permission from '../types/Perm';
-import { AuthenticatedRequest, Doc, TypedRequestBody } from '../types/UtilTypes';
+import { TypedRequestParams, TypedResponse } from '../types/UtilTypes';
 import { UserIdParam } from '../types/RequestParams';
 import { TypedRequest } from '../types/UtilTypes';
-import Role, { RoleModel } from '../models/RoleModel';
+import Role, { RoleDocument } from '../models/RoleModel';
 import GooglePayload from '../types/GooglePayload';
+import {
+    DevLoginRequest,
+    GiveRolesRequest,
+    LoginRequest,
+    RemoveRolesRequest,
+    UpdateUserRequest,
+} from '../shared/Types/requests/UserRequests';
+import {
+    GetAllUsersData,
+    GetSelfData,
+    GetUserData,
+    GiveRolesData,
+    LoginData,
+    RemoveRolesData,
+    UpdateUserData,
+} from '../shared/Types/responses/UserResponsesData';
+import Permission from '../shared/Types/utils/Permission';
+import { Request } from 'express';
 
 const client = new OAuth2Client(config.clientID);
 
 /**
  * @desc    An endpoint that is only accessible during development for getting a JWT token for the specified user id.
  * @route   POST api/users/devlogin
  */
-const devLoginUser = asyncHandler(async (req: TypedRequestBody<DevLoginRequest>, res: Response) => {
+const devLoginUser = asyncHandler(async (req: Request<DevLoginRequest>, res: TypedResponse<LoginData>) => {
```

Comment:
This is intentional, because you dont need to be logged in to use this request (Whereas TypedRequests store the requester so expect you to be logged in).

I have just realised that the login endpoint should be using Request too... I'll fix that up quickly